ios/FluentUI/Vnext/PersonaButton/PersonaButton.swift

```diff
@@ -0,0 +1,273 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import SwiftUI
+
+/// `MSFPersonaButtonState` contains PersonaButton properties in addition to a subset of the MSFAvatarState protocol.
+///
+/// `onTapAction` provides tap gesture for PersonaButton.
+///
+@objc public protocol MSFPersonaButtonState: MSFAvatarState {
+    var badgeSize: MSFPersonaButtonSize { get set }
+    var onTapAction: (() -> Void)? { get set }
+
+    var backgroundColor: UIColor? { get set }
+    var foregroundColor: UIColor? { get set }
+    var hasPointerInteraction: Bool { get set }
+    var hasRingInnerGap: Bool { get set }
+    var image: UIImage? { get set }
+    var imageBasedRingColor: UIImage? { get set }
+    var isOutOfOffice: Bool { get set }
+    var isRingVisible: Bool { get set }
+    var isTransparent: Bool { get set }
+    var presence: MSFAvatarPresence { get set }
+    var primaryText: String? { get set }
+    var ringColor: UIColor? { get set }
+    var secondaryText: String? { get set }
+}
+
+/// Properties that make up PersonaButton content
+class MSFPersonaButtonViewStateImpl: NSObject, ObservableObject, Identifiable, MSFPersonaButtonState {
+    @Published var badgeSize: MSFPersonaButtonSize
+    @Published var onTapAction: (() -> Void)?
+
+    let tokens: MSFPersonaButtonTokens
+
+    var backgroundColor: UIColor? {
+        get {
+            return avatarState.backgroundColor
+        }
+        set {
+            avatarState.backgroundColor = newValue
+            objectWillChange.send()
```

Comment:
I think you don't need to call it here. The avatarState implementation does it as a @Published property in an ObservableObject.

Reply:
That works for people observing `avatarState` directly, but this is necessary if you're expecting to get notifications against the wrapped property on `MSFPersonaButtonViewStateImpl`.

That said, some of these property names are confusing since they don't map directly to what the button does, so I'll rename them.

Reply:
I think SwiftUI will track that if the avatar View changes and is contained by this control.

The useful application of the objectWillChange.send() call I've seen so far is to force SwiftUI refreshes if you either don't have a @published property or you're implementing a custom property that will be observed.

These cases are just a passthrough to the @published properties of the avatar, which is what makes the objectWitllChange call not needed.

Reply:
Well that is just the darndest thing. I would have put serious money on those calls being necessary at some point during development, but you're right: they don't seem to do anything now. I'll get them removed as part of the next big PR for the carousel.